app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,221 @@
+
+  const setRecipeNameErrorData = (
+    recipeNameError: IRecipeNameErrorData,
+    formData: IRecipeFormData = recipeFormData
+  ) => {
+    setRecipeNameErrorDataState(recipeNameError);
+    handleSaveButtonMode(formData, recipeNameError);
+  };
+
+  const handleSaveButtonMode = (
+    formData: IRecipeFormData = recipeFormData,
+    nameErrorData: IRecipeNameErrorData = recipeNameErrorData
+  ) => {
+    const shouldDisableSaveButton =
+      formData.recipeName === '' ||
+      formData.description === '' ||
+      formData.recipeName?.trim().length === 0 ||
+      formData.description?.trim().length === 0 ||
+      nameErrorData.isRecipeNameError;
+    setIsSaveDisabled(shouldDisableSaveButton);
+  };
+
+  const handleRecipeFormData = (formData: IRecipeFormData) => {
+    setRecipeFormData(formData);
+    handleSaveButtonMode(formData);
+  };
+
+  const onRecipeNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    handleRecipeFormData({
+      ...recipeFormData,
+      recipeName: event.target.value,
+    });
+    validateIfRecipeNameExists.current({
+      recipeName: event.target.value,
+      description: recipeFormData.description,
+    });
+  };
+
+  const onRecipeDescriptionChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    handleRecipeFormData({
+      ...recipeFormData,
+      description: event.target.value,
+    });
+  };
+
+  const onFormSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setApiParams({
+      ...apiParams,
+      createRecipeParams: {
+        context: getCurrentNamespace(),
+      },
+    });
+  };
+
+  const onCancel = () => {
+    setShowRecipeForm(false);
+  };
+
+  /*
+   * In this function we are validating recipe name input filed
+   * (whether recipe name already exists or not and recipe name without alphanumeric characters)
+   * based on the result we are showing the helper text
+   */
+
+  const validateIfRecipeNameExists = useRef(
+    debounce((formData: IRecipeFormData) => {
+      if (formData.recipeName && !recipeNameRegEx.test(formData.recipeName)) {
+        setRecipeNameErrorData({
+          isRecipeNameError: true,
+          recipeNameErrorMessage: T.translate(`${PREFIX}.validationErrorMessage`).toString(),
+        });
+      } else {
+        if (formData.recipeName) {
+          setApiParams({
+            ...apiParams,
+            getRecipeByNameParams: {
+              context: getCurrentNamespace(),
+              recipeName: formData.recipeName,
+            },
+          });
+        } else {
+          setRecipeNameErrorData(noErrorState);
+        }
+      }
+    }, 500)
+  );
+
+  return (
+    <RecipeForm
+      recipeFormData={recipeFormData}
+      isRecipeNameError={recipeNameErrorData.isRecipeNameError}
+      recipeNameErrorMessage={recipeNameErrorData.recipeNameErrorMessage}
+      onRecipeNameChange={onRecipeNameChange}
+      onFormSubmit={onFormSubmit}
+      onCancel={onCancel}
```

Comment:
can we do `()=> setShowRecipeForm(false);` directly
if onCancel is called anywhere else??